pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -132,17 +126,12 @@ def advance(self, batch, batch_idx):
         self.trainer.logger_connector.on_train_split_start(batch_idx, split_idx, split_batch)
 
         if self.trainer.lightning_module.automatic_optimization:
-            for opt_idx, optimizer in self.get_active_optimizers(batch_idx):
-                # handle optimization restart
-                if self.restarting:
-                    if opt_idx < self.optim_progress.optimizer_idx:
-                        continue
-
-                self.optim_progress.optimizer_idx = opt_idx
-
-                result = self._run_optimization(batch_idx, split_batch, opt_idx, optimizer)
-                if result:
-                    self.batch_outputs[opt_idx].append(copy(result.result_collection))
+            # in automatic optimization, hand over execution to the OptimizerLoop
+            optimizers = [optimizer for _, optimizer in self.get_active_optimizers(batch_idx)]
+            batch_outputs, self._hiddens = self.optimizer_loop.run(split_batch, self._hiddens, optimizers, batch_idx)
+            # combine outputs from each optimizer
+            for k in range(len(batch_outputs)):
+                self.batch_outputs[k].extend(batch_outputs[k])
```

Comment:
Why do we extend now instead of append as in the previous version?

Reply:
there is an append inside the optimizer loop to it's own outputs list. 
here we need to extend to make it equivalent to the previous case where we had a single outputs list for the combined loop.

Reply:
happy to hear suggestions for how to simplify outputs handling across loops, but definitely out of scope for this PR.

Reply:
This is fine with me!